pkg/registration/supply_chain_factory.go

```diff
@@ -110,6 +124,53 @@ func (f *SupplyChainFactory) createSupplyChain() ([]*proto.TemplateDTO, error) {
 	return supplyChainBuilder.Create()
 }
 
+// Stitching metadata required for stitching with XL
+func (f *SupplyChainFactory) buildNodeMergedEntityMetadata() (*proto.MergedEntityMetadata, error) {
+	fieldsCapactiy := map[string][]string{
+		builder.PropertyCapacity: {},
+	}
+	fieldsUsedCapacity := map[string][]string{
+		builder.PropertyUsed:     {},
+		builder.PropertyCapacity: {},
+	}
+	fieldsUsedCapacityPeak := map[string][]string{
+		builder.PropertyUsed:     {},
+		builder.PropertyCapacity: {},
+		builder.PropertyPeak:     {},
+	}
+	mergedEntityMetadataBuilder := builder.NewMergedEntityMetadataBuilder()
+	// Set up matching criteria based on stitching type
+	switch f.stitchingPropertyType {
+	case stitching.UUID:
+		mergedEntityMetadataBuilder.
+			InternalMatchingType(builder.MergedEntityMetadata_STRING).
+			InternalMatchingProperty(proxyVMUUID).
+			ExternalMatchingType(builder.MergedEntityMetadata_STRING).
+			ExternalMatchingField(VMUUID, []string{})
+	case stitching.IP:
+		mergedEntityMetadataBuilder.
+			InternalMatchingType(builder.MergedEntityMetadata_LIST_STRING).
+			InternalMatchingPropertyWithDelimiter(proxyVMIP, ",").
+			ExternalMatchingType(builder.MergedEntityMetadata_LIST_STRING).
+			ExternalMatchingFieldWithDelimiter(VMIPFieldName, VMIPFieldPaths, ",")
+	default:
+		return nil, fmt.Errorf("stitching property type %s is not supported",
+			f.stitchingPropertyType)
+	}
+	return mergedEntityMetadataBuilder.
+		PatchSoldMetadata(proto.CommodityDTO_CLUSTER, fieldsCapactiy).
+		PatchSoldMetadata(proto.CommodityDTO_VMPM_ACCESS, fieldsCapactiy).
+		PatchSoldMetadata(proto.CommodityDTO_VCPU, fieldsUsedCapacityPeak).
+		PatchSoldMetadata(proto.CommodityDTO_VMEM, fieldsUsedCapacity).
```

Comment:
Now that we patch vcpu peak in https://github.com/turbonomic/kubeturbo/pull/255, shall we also patch vmem peak?

Reply:
Thanks @chlam4 , yes we should patch VMEM peak as well. We set peak as the used value for sold commodities [here](https://github.com/turbonomic/kubeturbo/blob/6e9a5a9108e569d046e7e8cb85b526ca92703701/pkg/discovery/dtofactory/general_builder.go#L156).